data/pom.xml

```diff
@@ -167,7 +167,7 @@
         <dependency>
             <groupId>org.apache.httpcomponents</groupId>
```

Comment:
Can we not remove the dep on apache http client?
Or even just some of the components?

Reply:
Not yet unfortunately, the URIBuilder that is used extensively in the Kusto library is in the client package and not the core package, so we have to keep everything for now. It was moved to the Apache HTTP core in version 5, and there are other URI Builder libraries out there that are very small footprint, however last time I tried updating to the Apache V5 URIBuilder it caused failures with those really weird URIs in the tests.

That is basically the only remaining Apache component from the http-client package that we use but it would require careful removal. I would like to remove it at a later date (soon).

Reply:
Moved all URIBuilder code into URIUtils to make it easier to rip out the Apache builder in the near future.

Reply:
So should we just move to http core version 5 then? any reason not to?

Reply:
Between V4 and V5 of the Apache URIBuilder either there was a regression or some of the URIs we're testing are invalid. If you upgrade to V5 most URIs in our test cases will pass however one or two fail.